glean-core/ffi/src/timing_distribution.rs

```diff
@@ -48,6 +50,28 @@ pub extern "C" fn glean_timing_distribution_cancel(metric_id: u64, timer_id: Tim
     })
 }
 
+#[no_mangle]
+pub extern "C" fn glean_timing_distribution_accumulate_samples(
+    glean_handle: u64,
+    metric_id: u64,
+    raw_samples: RawInt64Array,
+    num_samples: i32,
+) {
+    GLEAN.call_infallible(glean_handle, |glean| {
+        TIMING_DISTRIBUTION_METRICS.call_infallible_mut(metric_id, |metric| {
+            // The Kotlin code is sending Long(s), which are 64 bits, as there's
+            // currently no stable UInt type. The positive part of [Int] would not
+            // be enough to represent the values coming in:.
+            // Here Long(s) are handled as i64 and then casted to u32.
```

Comment:
The casting happens inside `accumulate_samples_signed` (after appropriate negativity checks). That should be made more clear in the comment.